packages/esm-system-admin-app/src/change-password-modal/change-password-modal.component.tsx

```diff
@@ -0,0 +1,197 @@
+    }
+    if (passwordInput.newPassword !== '') {
+      handleValidation(passwordInput.newPassword, 'newPassword');
+    }
+    if (passwordInput.confirmPassword !== '') {
+      handleValidation(passwordInput.confirmPassword, 'confirmPassword');
+    }
+  }, [handleValidation, passwordInput]);
+
+  const handlePasswordChange = (event) => {
+    const passwordInputValue = event.target.value.trim();
+    const passwordInputFieldName = event.target.name;
+    const NewPasswordInput = { ...passwordInput, [passwordInputFieldName]: passwordInputValue };
+    setPasswordInput(NewPasswordInput);
+  };
+
+  const handleSubmit = useCallback(
+    async (evt: React.FormEvent<HTMLFormElement>) => {
+      evt.preventDefault();
+      setIsSavingPassword(true);
+      performPasswordChange(passwordInput.oldPassword, passwordInput.confirmPassword)
+        .then(() => {
+          close();
+          navigate({ to: `\${openmrsSpaBase}/logout` });
+          showSnackbar({
+            isLowContrast: true,
+            kind: 'success',
+            subtitle: t('userPasswordUpdated', 'User password updated successfully'),
+            title: t('userPassword', 'User password'),
+          });
+        })
+        .catch((error) => {
+          setIsSavingPassword(false);
+          showSnackbar({
+            title: t('invalidPasswordCredentials', 'Invalid password provided'),
+            kind: 'error',
+            isLowContrast: false,
+            subtitle: error?.message,
+          });
+        });
+    },
+    [close, passwordInput.confirmPassword, passwordInput.oldPassword, t],
+  );
+
+  return (
+    <>
+      <ModalHeader closeModal={close} title={t('changePassword', 'Change Password')} />
+      <ModalBody>
+        <Tile>
+          <Form onSubmit={handleSubmit} ref={formRef}>
+            <div className={styles['input-group']}>
+              <Layer>
+                <PasswordInput
+                  id="oldPassword"
+                  invalid={oldPasswordError.length > 0}
+                  invalidText={oldPasswordError}
+                  labelText={t('oldPassword', 'Old Password')}
+                  name="oldPassword"
+                  value={passwordInput.oldPassword}
+                  onChange={handlePasswordChange}
+                  ref={oldPasswordInputRef}
+                  required
+                  showPasswordLabel="Show old password"
+                />
+              </Layer>
+              <Layer>
+                <PasswordInput
+                  id="newPassword"
+                  invalid={newPasswordError.length > 0}
+                  invalidText={newPasswordError}
+                  labelText={t('newPassword', 'New Password')}
+                  name="newPassword"
+                  value={passwordInput.newPassword}
+                  onChange={handlePasswordChange}
+                  ref={newPasswordInputRef}
+                  required
+                  showPasswordLabel="Show new password"
```

Comment:
Consider providing detailed helper text below the fields listing any requirements related to the data format, such as types of characters allowed per the password input [usage guidelines](https://carbondesignsystem.com/components/text-input/usage#password-input).